File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l,r = 0, len(numbers) - 1
        curr = numbers[l] + numbers[r]
        while curr != target:
            if curr > target:
                r -= 1
            else:
                l += 1
            curr = numbers[l] + numbers[r]
        return [l + 1, r + 1]
        #WCRT: O(N) | Space: O(1)

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> hashmap;
        for(int i=0; i<nums.size(); i++){
            if(hashmap[target - nums[i]] != 0){
                return {hashmap[target - nums[i]] - 1, i};
            }
            hashmap[nums[i]] = i + 1;
        }
    }
};
